Unable to find class `IntegrityChecker` in latest DANA version

### Application Version
BypassDANA: v1.1
DANA: v2.49.1
Xposed Framework: LSPosed

### Logcat
```
Caused by: java.lang.ClassNotFoundException: id.dana.utils.IntegrityChecker
	at java.lang.Class.classForName(Native Method) 
	at java.lang.Class.forName(Class.java:454) 
	at q.a(Unknown Source:16) 
	at VkDJa.UjoJulzYRkB.GWVN.XposedHelpers.findClass(Unknown Source:4) 
	at VkDJa.UjoJulzYRkB.GWVN.XposedHelpers.findAndHookMethod(SourceFile:2) 
	at aimardcr.co.id.bypass_dana.NotSoUnsafe.handleLoadPackage(NotSoUnsafe.java:25) 
	at VkDJa.UjoJulzYRkB.GWVN.IXposedHookLoadPackage$Wrapper.handleLoadPackage(Unknown Source:2) 
	at VkDJa.UjoJulzYRkB.GWVN.callbacks.XC_LoadPackage.call(Unknown Source:6) 
	at VkDJa.UjoJulzYRkB.GWVN.callbacks.XCallback.callAll(Unknown Source:26) 
	at g0.a(Unknown Source:281) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at J.callback(Unknown Source:250) 
	at LSPHooker_.createOrUpdateClassLoaderLocked(Unknown Source:11) 
	at android.app.LoadedApk.getClassLoader(LoadedApk.java:1126) 
	at android.app.LoadedApk.getResources(LoadedApk.java:1374) 
	at android.app.ContextImpl.createAppContext(ContextImpl.java:3102) 
	at android.app.ContextImpl.createAppContext(ContextImpl.java:3094) 
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6722) 
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2134) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:201) 
	at android.os.Looper.loop(Looper.java:288) 
	at android.app.ActivityThread.main(ActivityThread.java:7975) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:942) 
```

--- a/README.md
+++ b/README.md
@@ -1,6 +1,9 @@
 # Bypass DANA
 This Xposed module will bypass DANA's Integrity Check by hooking several security check functions.
 
+# Tested
+Currently Support until DANA v2.49.1
+
 # Installation
 1. Install Xposed (Alternative: EdXposed, LSPosed)
 2. Install the release `.apk`

--- a/app/src/main/java/aimardcr/co/id/bypass_dana/NotSoUnsafe.java
+++ b/app/src/main/java/aimardcr/co/id/bypass_dana/NotSoUnsafe.java
@@ -2,6 +2,9 @@ package aimardcr.co.id.bypass_dana;
 
 import android.app.Application;
 import android.content.Context;
+import android.content.ContextWrapper;
+import android.content.Intent;
+import android.os.Bundle;
 import android.widget.Toast;
 
 import androidx.annotation.NonNull;
@@ -12,6 +15,7 @@ import de.robv.android.xposed.XposedHelpers;
 import de.robv.android.xposed.callbacks.XC_LoadPackage;
 
 public class NotSoUnsafe implements IXposedHookLoadPackage {
+    String TAG = "NotSoUnsafe";
 
     // target app classloader & context
     private ClassLoader pClassLoader;
@@ -26,102 +30,34 @@ public class NotSoUnsafe implements IXposedHookLoadPackage {
         pClassLoader = lpparam.classLoader;
         initAppContext();
     }
+
+    private void intentHook() {
+        try {
+            XposedHelpers.findAndHookMethod(ContextWrapper.class, "startActivity", Intent.class, new XC_MethodHook() {
+                @Override
+                protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
+                    Intent intent = (Intent) param.args[0];
+                    Bundle bundle = intent.getExtras();
+                    if (bundle != null) {
+                        String str = bundle.getString("unsafe_status");
+                        if (str != null) {
+                            param.setResult(null);
+                        }
+                    }
+                }
+            });
+        } catch (Throwable e) {
+            e.printStackTrace();
+            Toast.makeText(context, "Unable to bypass detection!", Toast.LENGTH_SHORT).show();
+        }
+    }
+
     public void initHook() {
-        integrityCheck();
-        emuCheck();
-        hookCheck();
-        rootCheck();
-        tampCheck();
-    }
-
-    public void integrityCheck() {
         if (pClassLoader == null || context == null) {
             throw new NullPointerException("pClassLoader/context is null");
         }
 
-        try {
-            XposedHelpers.findAndHookMethod("id.dana.utils.IntegrityChecker", pClassLoader, "ArraysUtil", String.class, new XC_MethodHook() {
-                @Override
-                protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
-                    param.setResult(null);
-                }
-            });
-        } catch (Throwable e) {
-            e.printStackTrace();
-            Toast.makeText(context, "Unable to bypass integrityCheck!", Toast.LENGTH_SHORT).show();
-        }
-    }
-
-    public void emuCheck() {
-        if (pClassLoader == null || context == null) {
-            throw new NullPointerException("pClassLoader/context is null");
-        }
-
-        try {
-            XposedHelpers.findAndHookMethod("id.dana.DanaApplication", pClassLoader, "emuCb", long.class, long.class, new XC_MethodHook() {
-                @Override
-                protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
-                    param.setResult(null);
-                }
-            });
-        } catch (Throwable e) {
-            e.printStackTrace();
-            Toast.makeText(context, "Unable to bypass emuCheck!", Toast.LENGTH_SHORT).show();
-        }
-    }
-
-    public void hookCheck() {
-        if (pClassLoader == null || context == null) {
-            throw new NullPointerException("pClassLoader/context is null");
-        }
-
-        try {
-            XposedHelpers.findAndHookMethod("id.dana.DanaApplication", pClassLoader, "hookCb", long.class, long.class, new XC_MethodHook() {
-                @Override
-                protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
-                    param.setResult(null);
-                }
-            });
-        } catch (Throwable e) {
-            e.printStackTrace();
-            Toast.makeText(context, "Unable to bypass hookCheck!", Toast.LENGTH_SHORT).show();
-        }
-    }
-
-    public void rootCheck() {
-        if (pClassLoader == null || context == null) {
-            throw new NullPointerException("pClassLoader/context is null");
-        }
-
-        try {
-            XposedHelpers.findAndHookMethod("id.dana.DanaApplication", pClassLoader, "rootCb", long.class, long.class, new XC_MethodHook() {
-                @Override
-                protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
-                    param.setResult(null);
-                }
-            });
-        } catch (Throwable e) {
-            e.printStackTrace();
-            Toast.makeText(context, "Unable to bypass rootCheck!", Toast.LENGTH_SHORT).show();
-        }
-    }
-
-    public void tampCheck() {
-        if (pClassLoader == null || context == null) {
-            throw new NullPointerException("pClassLoader/context is null");
-        }
-
-        try {
-            XposedHelpers.findAndHookMethod("id.dana.DanaApplication", pClassLoader, "tampCb", long.class, long.class, new XC_MethodHook() {
-                @Override
-                protected void beforeHookedMethod(XC_MethodHook.MethodHookParam param) throws Throwable {
-                    param.setResult(null);
-                }
-            });
-        } catch (Throwable e) {
-            e.printStackTrace();
-            Toast.makeText(context, "Unable to bypass tampCheck!", Toast.LENGTH_SHORT).show();
-        }
+        intentHook();
     }
 
     public void initAppContext() {

